addons/website_event/static/src/snippets/options.js

```diff
@@ -10,19 +10,20 @@ options.registry.WebsiteEvent = options.Class.extend({
     async start() {
         const res = await this._super(...arguments);
         this.currentWebsiteUrl = this.ownerDocument.location.pathname;
-        this.eventId = this._getEventObjectId();
+        this.eventId = this._getEventObjectId(this.currentWebsiteUrl);
         // Only need for one RPC request as the option will be destroyed if a
         // change is made.
         const rpcData = await this._rpc({
             model: 'event.event',
             method: 'read',
             args: [
                 [this.eventId],
-                ['website_menu'],
+                ['website_menu', 'website_url'],
             ],
         });
-        this.data.reload = this.currentWebsiteUrl;
+        this.eventWebsiteUrl = rpcData[0]['website_url'];
```

Comment:
So this is part of my suggested solution, except that we will now always redirect to the event url, even for other options such as: https://github.com/odoo/odoo/blob/16.0/addons/website_event_meet/static/src/js/snippets/options.js#L40

Which is not what we want.

I don't really get your reply:
> this.data.reload isn't triggering a URL reload

From what I understand it's not a function, just something you set to a URL like you did below.

My solution was to set it to the website event URL **only when we modify the sub menus option**, and not the other options, as explained in the previous comment.